### DISCUSSION TITLE
Unable to initialize module when not using IncludeAllContentForSelfExtract, .net 5

### DISCUSSION THREAD

**Discussion by DanielRBaird** [cefsharp/CefSharp#3533] [Q&A]
I'm currently working on upgrading an application to .net 5, and I'm trying to take advantage of the changes to single file publishing so that we don't need to extract so many files. 

Unfortunately when using the latest netcore version of cefsharp this results in a crash unless you use `IncludeAllContentForSelfExtract`.

This can be reproduced in the minimal example, but removing the `IncludeAllContentForSelfExtract` setting and publishing a single file for net-5.0. 

The result is this crash:
```
Application: CefSharp.MinimalExample.WinForms.netcore.exe
CoreCLR Version: 5.0.521.16609
.NET Version: 5.0.5
Description: The process was terminated due to an unhandled exception.
Exception Info: System.TypeInitializationException: The type initializer for '<Module>' threw an exception.
 ---> System.ArgumentNullException: Value cannot be null. (Parameter 'path1')
   at System.IO.Path.Combine(String path1, String path2)
   at CefSharp.Initializer.ModuleInitializer()
   at .cctor()
   --- End of inner exception stack trace ---
   at CefSharp.MinimalExample.WinForms.ProgramPublishSingleFile.Main(String[] args)
```

Location of the crash: [Initializer.cs](https://github.com/cefsharp/CefSharp/blob/4608924d5a0c676857720917550674b88bd7b701/CefSharp.Core/Initializer.cs#L44)

Ultimately this is caused because net 5 returns an empty string for `Assembly.Location` now, so another method is needed to get the proper paths.

Should this be made into a bug issue? 
Has anyone figured another way around it without including all content for self extract?

**Comment by amaitland** [comment]
> Should this be made into a bug issue?

A PR referencing this discussion would be preferable.

> Has anyone figured another way around it without including all content for self extract?

If your embedding the native resources then I'd be surprised if this is possible though your welcome to do your own research, reach out to someone on the dot net team perhaps?  #3407 has my research to date 

If you don't embed the native resources then this should be possible (from memory this is the default).

**Reply by amaitland** [reply]
When `Assembly.Location` is empty we'll fallback to `AppContext.BaseDirectory`

Changes will be in the 90 release, commit https://github.com/cefsharp/CefSharp/commit/82daa955a26eebb2fd1f5427c41f1d58dc01e820

**Reply by DanielRBaird** [reply]
Cool, thanks for the info @amaitland.
